### PR TITLE
[#373] Add a workflow to automatically draft a release

### PR DESCRIPTION
**Pull request by ducbm051291**
- #373 

## What happened

- Create a workflow to generate a draft release automatically.
 
## Insight

- Create release-drafter.yml workflow file in .github/workflow folder
- Create release-drafter.yml template file in .github folder
 
## Proof Of Work

**[Trigger when Release PR merged into main](https://github.com/ducbm051291/surveys-ic/actions/runs/3546673464)**  

**[Result](https://github.com/ducbm051291/surveys-ic/releases)**

### CODE REVIEW

**Review by blyscuit — APPROVED** [review]

File: .github/release-drafter.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    labels:
+      - 'type: feature'
+      - 'type: chore'
+  patch:
+    labels:
+      - 'type: bug'
+  default: minor
+  
+template: |
+  ## What's Changed
```
Comment (suho) [inline .github/release-drafter.yml:26]: We don't need this section title, just need
```
template: |
  $CHANGES
```

File: .github/workflows/release-drafter.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    # branches to consider in the event; optional, defaults to all
+    branches:
+      - main
+  # pull_request event is required only for autolabeler
+  pull_request:
+    # Only following types are handled by the action, but one can default to all as well
+    types: [opened, reopened, synchronize]
+  # pull_request_target event is required for autolabeler to support PRs from forks
+  # pull_request_target:
+  #   types: [opened, reopened, synchronize]
```
Comment (suho) [inline .github/workflows/release-drafter.yml:14]: We should only have the main branch only, please remove `pull_request`

```suggestion
on:
  push:
    branches:
      - main
```

File: .github/workflows/release-drafter.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+name: Release Drafter
```
Comment (suho) [inline .github/workflows/release-drafter.yml:1]: The name should not `release drafter`. I prefer `Draft a new release`

Comment (suho) [reply .github/workflows/release-drafter.yml:1]: Please change the name of the file as well

File: .github/workflows/release-drafter.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  # pull_request_target:
+  #   types: [opened, reopened, synchronize]
+
+permissions:
+  contents: read
+
+jobs:
+  update_release_draft:
+    permissions:
+      # write permission is required to create a github release
```
Comment (suho) [inline .github/workflows/release-drafter.yml:22]: Please remove comments, I think we don't need it :D

File: .github/workflows/release-drafter.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      contents: write
+      # write permission is required for autolabeler
+      # otherwise, read permission is required at least
+      pull-requests: write
```
Comment (suho) [inline .github/workflows/release-drafter.yml:26]: We don't use `autolabeler`, so I think we don't need this one

**Review by suho — CHANGES_REQUESTED** [review]

**Review by nkhanh44 — APPROVED** [review]

File: .github/release-drafter.yml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+name-template: '$RESOLVED_VERSION'
+tag-template: '$RESOLVED_VERSION'
+categories:
+  - title: 'Features'
+    labels:
+      - 'type: feature'
```
Comment (vnntsu) [inline .github/release-drafter.yml:6]:
```suggestion
      - 'type : feature'
```

File: .github/release-drafter.yml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  - title: 'Chores'
+    label:
+      - 'type: chore'
```
Comment (vnntsu) [inline .github/release-drafter.yml:9]:
```suggestion
      - 'type : chore'
```

File: .github/release-drafter.yml [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  - title: 'Bugs'
+    labels:
+      - 'type: bug'
```
Comment (vnntsu) [inline .github/release-drafter.yml:12]:
```suggestion
      - 'type : bug'
```

Catch me if I'm wrong about the type labels

**Review by vnntsu — COMMENTED** [review]
Please check my commnets

File: .github/release-drafter.yml (hunk 8 above)
Comment (ducbm051291) [reply .github/release-drafter.yml:12]: @vnntsu it has to be the same as we configure our label. Seem like ios-template has space after type, let me update this.

Screenshot 2022-11-28 at 09 54 28

**Review by vnntsu — APPROVED** [review]
LGTM 🎉

**Review by Shayokh144 — APPROVED** [review]

**Review by suho — APPROVED** [review]

**Pull request merged**